src/BoolUnification.js

```diff
@@ -117,6 +128,70 @@ function satisfiable(f) {
     }
 }
 
+/**
+ * Check whether an equation is of the form `Var(X) =? Equation`, where X not in free(Equation).
+ */
+function isSingleVarSpecialCase(maybeVar, maybeAssigned) {
+    return isVar(maybeVar) && !boolFreeVars(maybeAssigned).includes(maybeVar.name);
+}
+
```

Comment:
Add comment with a bit of intuition. At minimum explain what the function returns.